pmlc/conversion/linalg_to_pxa/linalg_to_pxa.cc

```diff
@@ -0,0 +1,503 @@
+      agg = AtomicRMWKind::muli;
+    } else if (matchPattern(op,
+                            m_Op<MulFOp>(m_Capture(&lhs), m_Capture(&rhs)))) {
+      agg = AtomicRMWKind::mulf;
+    } else if (matchPattern(
+                   op, m_Op<SelectOp>(
+                           m_Capture(&cond, m_Op<CmpIOp>(m_Capture(&lhs),
+                                                         m_Capture(&rhs))),
+                           m_Capture(&trueValue), m_Capture(&falseValue)))) {
+      if (lhs == trueValue && rhs == falseValue) {
+        relatedOp = cond.getDefiningOp();
+        CmpIPredicate intPred = cast<CmpIOp>(relatedOp).predicate();
+        switch (intPred) {
+        case CmpIPredicate::sgt:
+        case CmpIPredicate::sge:
+          agg = AtomicRMWKind::maxs;
+          break;
+        case CmpIPredicate::ugt:
+        case CmpIPredicate::uge:
+          agg = AtomicRMWKind::maxu;
+          break;
+        case CmpIPredicate::slt:
+        case CmpIPredicate::sle:
+          agg = AtomicRMWKind::mins;
+          break;
+        case CmpIPredicate::ult:
+        case CmpIPredicate::ule:
+          agg = AtomicRMWKind::minu;
+          break;
+        default:
+          op->emitError("Invalid integer cmp predicate for aggregation.");
+        }
+      }
+    } else if (matchPattern(
+                   op, m_Op<SelectOp>(
+                           m_Capture(&cond, m_Op<CmpFOp>(m_Capture(&lhs),
+                                                         m_Capture(&rhs))),
+                           m_Capture(&trueValue), m_Capture(&falseValue)))) {
+      if (lhs == trueValue && rhs == falseValue) {
+        relatedOp = cond.getDefiningOp();
+        CmpFPredicate floatPred = cast<CmpFOp>(relatedOp).predicate();
+        switch (floatPred) {
+        case CmpFPredicate::OGT:
+        case CmpFPredicate::OGE:
+        case CmpFPredicate::UGT:
+        case CmpFPredicate::UGE:
+          agg = AtomicRMWKind::maxf;
+          break;
+        case CmpFPredicate::OLT:
+        case CmpFPredicate::OLE:
+        case CmpFPredicate::ULT:
+        case CmpFPredicate::ULE:
+          agg = AtomicRMWKind::minf;
+          break;
+        default:
+          op->emitError("Invalid float cmp predicate for aggregation.");
+        }
+      }
+    } else {
+      op->emitError("Invalid operation(s) for reduction.");
+    }
+    scalar = (lhs == bufElem) ? rhs : lhs;
+  }
+
+  AtomicRMWKind getAgg() { return agg; }
+  Value getScalar() { return scalar; }
+  Operation *getRelatedOp() { return relatedOp; }
+
+private:
+  AtomicRMWKind agg;
+  Value scalar;
+  Operation *relatedOp;
+};
+
+struct ConvOpConversion : public OpConversionPattern<ConvOp> {
+  using OpConversionPattern<ConvOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ConvOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    op.emitError("The program should not contain linalg.conv");
```

Comment:
For all of the cases where we are adding the 'Generalization' patterns (conv, padding, etc), we should probably just remove these patterns.  Since those ops are not in the 'legal' set, if they are not generalized, the pass will fail with a nice error saying that conv is not a legal op.

Reply:
Removed these rewriters, and added them in the illegal op list.